Assets/Scripts/Player/Controller/WeaponSwitchController.cs

```diff
@@ -0,0 +1,154 @@
+        
+        weapons.Add(_shooter.getFireWeapon().gameObject);
+        currentWeapon = _shooter.getFireWeapon().gameObject;
+
+        _cameraTransform = _shooter.getPoV();
+
+        SelectWeapon();
+    }
+
+    void Update()
+    {
+        int previousWeapon = selectedWeapon;
+
+        if (Input.GetAxis("Mouse ScrollWheel") > 0f && weapons.Count > 1)
+        {
+            if (selectedWeapon >= transform.childCount - 1)
+                selectedWeapon = 0;
+            else
+                selectedWeapon++;
+        }
+        else if (Input.GetAxis("Mouse ScrollWheel") < 0f && weapons.Count > 1)
+        {
+            if (selectedWeapon <= 0)
+                selectedWeapon = transform.childCount - 1;
+            else
+                selectedWeapon--;
+        }
+        else if (Input.GetKeyDown(KeyCode.Q) && currentWeapon) {
+            DropWeapon();
+        }
+
+        // If selected weapon changed, call selectweapon method to perform the weapon change
+        if (previousWeapon != selectedWeapon)
+            SelectWeapon();
+
+        RaycastHit hit;
+        Ray ray = new Ray(_cameraTransform.position, _cameraTransform.forward);
+        if (Physics.Raycast(ray, out hit))
+        {
+            if(hit.transform.CompareTag("FireWeapon")) {
+                FireWeapon weapon = hit.transform.gameObject.GetComponent<FireWeapon>();
+                if (Input.GetKeyDown(KeyCode.E) && weapons.Count < maxWeapons && !weapon.isInUse())
+                {            
+                    PickWeapon(_cameraTransform.GetChild(0), hit.collider);
+                }
+            }
+        }
+    }
+
+    void SelectWeapon()
+    {
+        int i = 0;
+        foreach (GameObject weapon in weapons)
+        {
+            // if the weapon is selected set true, if it's not set false
+            weapon.SetActive(i == selectedWeapon);
+            if (i == selectedWeapon)
+            {
+                FireWeapon fWeapon = weapon.GetComponent<FireWeapon>();
+                _shooter.setFireWeapon(fWeapon);
+                currentWeapon = weapon;
+            }
+            i++;
+        }
+    }
+
+    void PickWeapon(Transform container, Collider coll)
+    {
+        if(weapons.Count == 0 ){
+            _shooter.setFireWeapon(coll.gameObject.GetComponent<FireWeapon>());
+            currentWeapon = coll.gameObject;
+        }
+
+        // hides the weapon because it's now in our 'inventory'
+        coll.gameObject.SetActive(weapons.Count == 0);
+
+        // save the weapon                
+        weapons.Add(coll.gameObject);
+
+        coll.gameObject.GetComponent<FireWeapon>().setInUse(true);
+
+        if (weapons.Count == maxWeapons) slotFull = true;
+
+        //Make weapon a child of the camera and move it to default position
+        coll.transform.SetParent(container);
+        coll.transform.localPosition = Vector3.zero;
+        coll.transform.localRotation = Quaternion.Euler(0, 0, 0);
```

Comment:
The M4 is placed in a weird position. Bernas suggested the use of a transform in every weapon that represents the "handle" point. That way we could defined where the "zero" of each weapon is. What do you think? Maybe not for this PR, but we could create an issue or smth...

Reply:
Yes, we are waiting for the player model so that we can implement that